File: siserp-logistica-frontend/src/main/siserp-logistica-frontend/src/app/modules/intranet/services/marca.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { MarcaEliminarRequest } from '../dto/request/marca-eliminar.request';
import { MarcaListarRequest } from '../dto/request/marca-listar.request';
import { MarcaModificarRequest } from '../dto/request/marca-modificar.request';
import { MarcaRegistrarRequest } from '../dto/request/marca-registrar.request';
import { MarcaListarResponse } from '../dto/response/marca-listar.response';
import { MarcaRegistrarResponse } from '../dto/response/marca-registrar.response';
import { OutResponse } from '../dto/response/out.response';

@Injectable()
export class MarcaService {

  constructor(private http: HttpClient) { }

  public listarMarca(req: MarcaListarRequest): Observable<OutResponse<MarcaListarResponse[]>> {
    return this.http.post<OutResponse<MarcaListarResponse[]>>(`${environment.logisticaBackendUrl}/marca/listarMarca`, req);
  }

  public registrarMarca(req: MarcaRegistrarRequest): Observable<OutResponse<MarcaRegistrarResponse>> {
    return this.http.post<OutResponse<MarcaRegistrarResponse>>(`${environment.logisticaBackendUrl}/marca/registrarMarca`, req);
  }

  public modificarMarca(req: MarcaModificarRequest): Observable<OutResponse<any>> {
    return this.http.post<OutResponse<any>>(`${environment.logisticaBackendUrl}/marca/modificarMarca`, req);
  }

  public eliminarMarca(req: MarcaEliminarRequest): Observable<OutResponse<any>> {
    return this.http.post<OutResponse<any>>(`${environment.logisticaBackendUrl}/marca/eliminarMarca`, req);
  }
}

File: siserp-logistica-backend/src/main/java/com/besoft/siserp/lgtc/service/MarcaService.java
package com.besoft.siserp.lgtc.service;

import java.util.List;

import com.besoft.siserp.lgtc.dto.OutResponse;
import com.besoft.siserp.lgtc.dto.request.MarcaEliminarRequest;
import com.besoft.siserp.lgtc.dto.request.MarcaListarRequest;
import com.besoft.siserp.lgtc.dto.request.MarcaModificarRequest;
import com.besoft.siserp.lgtc.dto.request.MarcaRegistrarRequest;
import com.besoft.siserp.lgtc.dto.response.MarcaListarResponse;
import com.besoft.siserp.lgtc.dto.response.MarcaRegistrarResponse;

public interface MarcaService {

	public OutResponse<MarcaRegistrarResponse> registrarMarca(MarcaRegistrarRequest c);

	public OutResponse<?> modificarMarca(MarcaModificarRequest c);

	public OutResponse<List<MarcaListarResponse>> listarMarca(MarcaListarRequest c);
	
	public OutResponse<?> eliminarMarca(MarcaEliminarRequest c);
}
